client/components/domains/register-domain-step/index.jsx

```diff
@@ -844,22 +844,22 @@ class RegisterDomainStep extends Component {
 
 	fetchDomainPricePromise = ( domain ) => {
 		return new Promise( ( resolve ) => {
-			wpcom.undocumented().getDomainPrice( domain, ( serverError, result ) => {
-				if ( serverError ) {
+			wpcom.req
```

Comment:
You can remove the `new Promise` stuff and return a promise directly:
```js
return wpcom.req.get().then( data = > ( { ... } ) ).catch( error => ( { ... } ) )
```

The method could be named `fetchDomainPrice`, and when called on line 677, it doesn't need the `.catch` handler as it never throws.

Reply:
Happy to further clean up 👍  Done in https://github.com/Automattic/wp-calypso/pull/58902/commits/c4a9ef621187a634a5b567ee56713b73189dbc49 and https://github.com/Automattic/wp-calypso/pull/58902/commits/45b7eb944788c0d3d58727661dd6ee5a6ea1c492